tests/build-tests/components.test.ts

```diff
@@ -0,0 +1,106 @@
+import { execa } from 'execa';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { afterEach, beforeEach, describe, it } from 'vitest';
+
+import { AddonFixtureHelper } from '../fixture-helper.js';
+import { createAddon, createTmp, install } from '../utils.js';
+
+let packageManager = 'pnpm';
+
+describe('components (build)', () => {
+  describe('co-located JS', () => {
+    let cwd = '';
+    let tmpDir = '';
+    let addonOnlyJS: AddonFixtureHelper;
+
+    beforeEach(async () => {
+      tmpDir = await createTmp();
+
+      console.debug(`Debug test repo at ${tmpDir}`);
+
+      let { name } = await createAddon({
+        args: [`--pnpm=true`, '--addon-only'],
+        options: { cwd: tmpDir },
+      });
+
+      cwd = path.join(tmpDir, name);
```

Comment:
Nit, and unrelated to this PR really, but as I'm seeing this code here... `createAddon` receives `tmpDir`, and returns `name`, so has everything to compute `cwd`, so could we do this to simplify this a bit?

```suggestion
      let { name, addonDir: cwd } = await createAddon({
        args: [`--pnpm=true`, '--addon-only'],
        options: { cwd: tmpDir },
      });
```